examples/vision/attention_mil_classification.py

```diff
@@ -0,0 +1,578 @@
+image(s) will be divided and the subimages will form the bag of instances.
+
+Therefore, the goals are to:
+
+1. learn a model to predict class label of a bag of instances.
+2. know the score of the instances within the bag which resulted to the class label
+prediction.
+
+The second goal is of high interest because it will incorporate interpretability to
+the MIL approach. Such interpretability is deduced by using attention scores or weights
+as it will not merely predict the final output class but the instance scores that led
+to those classification results.
+
+## Implementation
+
+The MIL classifier is modelled using neural networks. The attention mechanism as MIL
+pooling:
+
+- Trainable MIL operator.
+- Interpretable results using attention scores.
+
+In the paper, both [histopathology](https://www.kaggle.com/paultimothymooney/breast-histopathology-images)
+and MNIST data were experimented with. In this example, we will experiment with MNIST
+data which can be simply called by `keras.datasets.mnist`.
+
+We will begin by creating the datasets (both, train and validation sets) and visualizing
+a sample of the dataset. Then we will build and train the model. Finally we will evaluate,
+check and visualize the classification results as well as the attention scores of the
+instances.
+
+Besides the implementation of this mechanism, some **regularization** techniques,
+**ensemble averaging** (for stability) and dealing with **imbalanced data** will be covered.
+
+## References
+
+- [Attention-based Deep Multiple Instance Learning](https://arxiv.org/pdf/1802.04712.pdf).
+- Some of attention operator code implementation was inspired from https://github.com/utayao/Atten_Deep_MIL.
+- Imbalanced data [tutorial](https://www.tensorflow.org/tutorials/structured_data/imbalanced_data)
+by TensorFlow.
+"""
+"""
+## Setup
+"""
+
+import numpy as np
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+from tqdm import tqdm
+from matplotlib import pyplot as plt
+
+plt.style.use("ggplot")
+
+"""
+## Create dataset
+At first we will set up the configurations and then prepare the datasets.
+
+We will create a set of bags and assign their labels accordingly. This is performed by choosing
+a positive instance and forming associated bags of instances. If atleast one positive instance
+is available in a bag, the bag is considered as a positive bag. If it does not contain any
+positive instance, the bag will be considered as negative.
+
+### Configurations
+
+- `POS_CLASS`: The desired class to be kept in the positive bag.
+- `BAG_COUNT`: The number of training bags.
+- `VAL_BAG_COUNT`: The number of validation bags.
+- `BAG_SIZE`: The number of instances in a bag.
+- `PLOT_SIZE`: The number of bags to plot.
+- `ENSEMBLE_AVG_COUNT`: The number of models to create and get their average. (OPTIONAL:
+often results in better performance - set to 1 for single model)
+"""
+
+POS_CLASS = 1
```

Comment:
Prefer using fully-spelled out variable names unless the abbreviation is very common, e.g. POSITIVE